src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -5809,11 +5813,42 @@ def run_network_configuration_diagnostic(cmd, client, watcher_rg, watcher_name,
 # endregion
 
 
+# region CustomIpPrefix
+def create_custom_ip_prefix(cmd, client, resource_group_name, custom_ip_prefix_name, location=None,
+                            cidr=None, tags=None, zone=None, signed_message=None, authorization_message=None,
+                            custom_ip_prefix_parent=None):
+
+    CustomIpPrefix = cmd.get_models('CustomIpPrefix')
+    prefix = CustomIpPrefix(
+        location=location,
+        cidr=cidr,
+        zones=zone,
+        tags=tags,
+        signed_message=signed_message,
+        authorization_message=authorization_message
+    )
+
+    if custom_ip_prefix_parent:
+        try:
+            prefix.custom_ip_prefix_parent = client.get(resource_group_name, custom_ip_prefix_name)
+        except ResourceNotFoundError:
+            raise ResourceNotFoundError("Custom ip prefix parent {} is'nt exist".format(custom_ip_prefix_name))
+
+    return client.begin_create_or_update(resource_group_name, custom_ip_prefix_name, prefix)
+
+
+def update_custom_ip_prefix(instance, tags=None):
+    if tags is not None:
+        instance.tags = tags
+    return instance
```

Comment:
Why this function only support tags? Can user update other parameters used in create command?

Reply:
just follow the powershell: https://docs.microsoft.com/en-us/powershell/module/az.network/update-azcustomipprefix?view=azps-6.2.1

Reply:
Please add other arguments which can be updated as well.